synthesizer/src/block/header/metadata/mod.rs

```diff
@@ -89,6 +94,8 @@ impl<N: Network> Metadata<N> {
                     && self.coinbase_target >= N::GENESIS_COINBASE_TARGET
                     // Ensure the proof target is at or above the minimum.
                     && self.proof_target >= N::GENESIS_PROOF_TARGET
```

Comment:
You can also add a check that `self.proof_target < self.coinbase_target` just to make sure order of arguments didn't mess up.